service/src/main/java/uk/nhs/adaptors/gp2gp/common/configuration/Gp2gpConfiguration.java

```diff
@@ -5,10 +5,12 @@
 
 import lombok.Getter;
 import lombok.Setter;
+import org.springframework.scheduling.annotation.EnableScheduling;
 
 @Getter
 @Setter
 @Configuration
+@EnableScheduling
```

Comment:
I would separate this out into it's own configuration - although not required, it helps adhere to the single responsibility principle.

Example of how I usually set it up in other projects:

```java
@Configuration
@EnableScheduling
class ScheduleConfiguration {
 // Empty for now, though would allow for scheduling specific bean registrations.
}
```

Reply:
as there were no other beans and configs defined in this place hence I thought that it wouldn't add a big burden if I add this annotation here. As this annotation is declarative and there are no many other such annotations for GP2GPConfiguration.java class I think I'd leave it here for the time being but a good point to watch it in the future if there appear more scheduler-related things.

Reply:
That's fine, I still think maybe it's better suited in a separate class - **S**OLID!  😄